src/components/button/button.js

```diff
@@ -138,24 +160,22 @@ function MdButtonDirective($mdButtonInkRipple, $mdTheming, $mdAria, $timeout) {
       }
     });
 
-    if (!angular.isDefined(attr.mdNoFocusStyle)) {
-      // restrict focus styles to the keyboard
-      scope.mouseActive = false;
-      element.on('mousedown', function() {
-        scope.mouseActive = true;
-        $timeout(function(){
-          scope.mouseActive = false;
-        }, 100);
-      })
-      .on('focus', function() {
-        if (scope.mouseActive === false) {
-          element.addClass('md-focused');
-        }
-      })
-      .on('blur', function(ev) {
-        element.removeClass('md-focused');
-      });
-    }
+    // restrict focus styles to the keyboard
+    scope.mouseActive = false;
+    element.on('mousedown', function() {
+      scope.mouseActive = true;
+      $timeout(function(){
+        scope.mouseActive = false;
+      }, 100);
+    })
+    .on('focus', function() {
+      if (scope.mouseActive === false) {
+        element.addClass('md-focused');
```

Comment:
No. Only add the `md-focused` class if the `md-no-focus` is **not** present. This will then let the CSS be simpler. 

> Fix the CSS above.